src/sentry/incidents/logic.py

```diff
@@ -791,6 +799,28 @@ def update_alert_rule(
                 DEFAULT_ALERT_RULE_RESOLUTION,
             )
 
+        if triggers is not None:
+            # Delete triggers we don't have present in the updated data.
+            trigger_ids = [x["id"] for x in triggers]
```

Comment:
I think this can error when we're creating new rows, since they won't have an `id`.